internal/orchestrator/updates.go

```diff
@@ -302,67 +328,87 @@ func (o *Orchestrator) eventUpdate(ctx context.Context, evt *v1types.ConditionUp
 		return errors.Wrap(err, "updating condition")
 	}
 
-	if rctypes.StateIsComplete(evt.ConditionUpdate.State) {
-		// deal with the completed event
-		delErr := status.DeleteCondition(evt.Kind, o.facility, evt.ConditionUpdate.ConditionID.String())
-		if delErr != nil {
-			// if we fail to delete this event from the KV, the reconciler will catch it later
-			// and walk this code, so return early.
-			o.logger.WithError(delErr).WithFields(logrus.Fields{
-				"condition.id":   evt.ConditionUpdate.ConditionID,
-				"server.id":      evt.ConditionUpdate.ServerID,
-				"condition.kind": evt.Kind,
-			}).Warn("removing completed condition data")
-			return errors.Wrap(errCompleteEvent, delErr.Error())
-		}
-		metrics.ConditionCompleted.With(
-			prometheus.Labels{
-				"conditionKind": string(evt.Kind),
-				"state":         string(evt.ConditionUpdate.State),
-			},
-		).Inc()
+	// nothing else to do if the condition is not finalized
+	if !rctypes.StateIsComplete(evt.ConditionUpdate.State) {
+		return nil
+	}
 
-		// queue any follow-on work as required
-		active, err := o.repository.GetActiveCondition(ctx, evt.ConditionUpdate.ServerID)
-		if err != nil && errors.Is(err, store.ErrConditionNotFound) {
-			// nothing more to do
-			return nil
-		}
+	// deal with the completed event
+	delErr := status.DeleteCondition(evt.Kind, o.facility, evt.ConditionUpdate.ConditionID.String())
+	if delErr != nil {
+		// if we fail to delete this event from the KV, the reconciler will catch it later
+		// and walk this code, so return early.
+		o.logger.WithError(delErr).WithFields(logrus.Fields{
+			"condition.id":   evt.ConditionUpdate.ConditionID,
+			"server.id":      evt.ConditionUpdate.ServerID,
+			"condition.kind": evt.Kind,
+		}).Warn("removing completed condition data")
+
+		return errors.Wrap(errCompleteEvent, delErr.Error())
+	}
+
+	metrics.ConditionCompleted.With(
+		prometheus.Labels{
+			"conditionKind": string(evt.Kind),
+			"state":         string(evt.ConditionUpdate.State),
+		},
+	).Inc()
+
+	// queue any follow-on work as required
+	return o.queueFollowingCondition(ctx, evt)
+}
+
+// Queue up follow on conditions
+//
+// TODO: Q. how do we know the follow on work is ordered as expected, or is the assumption that there could be only one other condition queued for this server ?
+// TODO: Q. if the current condition has failed, we most likely don't want to queue the next condition?
```

Comment:
If a condition has failed, when we update the store (cf. line 327) we will mark the entire `conditionRecord` as failed. When we call `o.repository.GetActiveCondition` that will then return `ErrConditionNotFound`, signifying that there is nothing more to do, and we short-circuit at line 369.

That logic for `GetActiveCondition` is [here](https://github.com/metal-toolbox/conditionorc/blob/16100161c40dc89f0543305a4069cca5aeaa5eeb/internal/store/nats.go#L109) while the update is [here](https://github.com/metal-toolbox/conditionorc/blob/16100161c40dc89f0543305a4069cca5aeaa5eeb/internal/store/nats.go#L227) and the failure case is explicitly [tested](https://github.com/metal-toolbox/conditionorc/blob/16100161c40dc89f0543305a4069cca5aeaa5eeb/internal/store/nats_test.go#L231)

Reply:
Thanks for explaining that, the linked test helped answer my questions as well.